### ISSUE TITLE
Ensure all deps are loaded over IPFS

### ISSUE THREAD

**Issue by olizilla** [ipfs-shipyard/ipfs-redux-bundle#2]
Right now `window.ipfs-fallback` loads `js-ipfs` or `js-ipfs-api` from https://unpkg.com. We want to ensure that all our deps are loaded via IPFS.

In IPLD Explorer https://github.com/ipfs-shipyard/ipld-explorer/blob/ff64b2fba5ad556af83617a488dc5a7b71aad572/src/bundles/ipfs.js I've pulled the ipfs redux bundle back inline, and used dynamic `import()` calls to selectively load `js-ipfs-api` or `js-ipfs`. Under the hood it's create-react-app and webpack, that manage splitting the js bundles where it finds calls to `import()` in code.

It'd be ideal if we could use that same logic here. I haven't checked if the webpack config used in create-react-app will transpile calls to `import()` from dependencies. Beyond that it'd be great to make this module useable by any build system, but as there isn't a huge amount of code here, it might be simpler to create new modules for different bundling mechanisms.

Alternatives are
- get `js-ipfs-api` and `js-ipfs` modules published to ipfs as part of the release process, and use those addresses in place of unpkg (@achingbrain is working on npm-on-ipfs as a much bigger effort to share npm modules over IPFS)
- ?

**Comment by olizilla** [comment]
Of note, in IPLD Explorer, we're not using window.ipfs-fallback so that we can control both the way that the deps are loaded (via webpack generated bundles from `import()` boundaries) and so we can do a multi step fallback from `window.ipfs` to `js-ipfs-api` to `js-ipfs`.

This was to ensure the ipld explorer works as well as it can in any set up. I think it's reasonable to do that here too, but we also need to think about what IPFS Web UI needs. There we want to try window.ipfs and fallback to `js-ipfs-api`, and if that fails, we want to prompt the user to configure a custom api address or ask if they'd like to use an embedded js-ipfs instance. It doesn't make sense in webui to auto fallback to an embedded js-ipfs instance, as that could lead to suprising behaviour when you are running a local daemon and then it stops for any reason.

**Comment by olizilla** [comment]
TL;DR
- How best to manage dynamic imports of js-ipfs-api and js-ipfs (imports() vs window.ipfs-fallback and loading ipfs modules over ipfs)
- How best to manage the order of the fallbacks. Many apps will want to use "the best ipfs provider available (window, ipfs-api, js-ipfs)", while web UI wants to use a specific IPFS provider and be very clear to the user which one it's using, and offer the ability to switch.

**Comment by olizilla** [comment]
ipfs-redux-bundle now depends directly on `js-ipfs-api` and the calling code can pass a version of`js-ipfs` to it if it's needed.
